Sources/Commands/Workspace.swift

```diff
@@ -281,22 +284,15 @@ public class Workspace {
         let rootManifest = try manifestLoader.load(packagePath: rootPackagePath, baseURL: rootPackagePath.asString, version: nil)
 
         // Compute the transitive closure of available dependencies.
-        let dependencies = transitiveClosure([KeyedPair(rootManifest, key: rootManifest.url)]) { node in
-            return node.item.package.dependencies.flatMap{ dependency in
+        let dependencies = try transitiveClosure([KeyedPair(rootManifest, key: rootManifest.url)]) { node in
+            return try node.item.package.dependencies.flatMap{ dependency in
                 // Check if this dependency is available.
                 guard let managedDependency = dependencyMap[RepositorySpecifier(url: dependency.url)] else {
                     return nil
                 }
 
                 // If so, load its manifest.
-                //
-                // This should *never* fail, because we should only have ever
-                // got this checkout via loading its manifest successfully.
-                //
-                // FIXME: Nevertheless, we should handle this failure explicitly.
-                //
-                // FIXME: We should have a cache for this.
-                let manifest: Manifest = try! manifestLoader.load(packagePath: checkoutsPath.appending(managedDependency.subpath), baseURL: managedDependency.repository.url, version: managedDependency.currentVersion)
+                let manifest: Manifest = try loadManifest(packagePath: checkoutsPath.appending(managedDependency.subpath), baseURL: managedDependency.repository.url, version: managedDependency.currentVersion)
```

Comment:
I think it would be better to fix this by trying to fix the core issue (which is that we should find a way to get the _actual_ manifest that was previously loaded).

The problem with cleaning this up is that IMHO it ends up obscuring that what the code is doing is actually *wrong*. I find that more obvious when the code has things like `try!` and `FIXMEs` littered around.

Note that the my comment about "we should have a cache for this" is probably wrong -- I think in this case what we want is to guarantee that we get the actual manifest that was read, not just happen to get the same manifest by going through a cache. This is a subtle, but important distinction: when the code doesn't need to use a cache to ensure it gets the same manifest then it ends up being more "solid" -- it becomes an invariant that this code operates on the same data that was previously seen, rather than an implicit dependency on the cache state.